VideoWeb/VideoWeb/ClientApp/src/app/services/conference/conference.service.ts

```diff
@@ -0,0 +1,164 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRoute, NavigationEnd, ParamMap, Router } from '@angular/router';
+import { Guid } from 'guid-typescript';
+import { BehaviorSubject, Observable, ReplaySubject, Subscription } from 'rxjs';
+import { filter, map, mergeMap, take } from 'rxjs/operators';
+import { ParticipantModel } from 'src/app/shared/models/participant';
+import { ApiClient, ConferenceResponse, ConferenceStatus } from '../clients/api-client';
+import { EventsService } from '../events.service';
+import { LoggerService } from '../logging/logger.service';
+import { ConferenceStatusMessage } from '../models/conference-status-message';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class ConferenceService {
+    private loggerPrefix = '[ConferenceService] -';
+
+    private subscriptions: Subscription[] = [];
+    constructor(
+        router: Router,
+        private activatedRoute: ActivatedRoute,
+        private eventService: EventsService,
+        private apiClient: ApiClient,
+        private logger: LoggerService
+    ) {
+        logger.conferenceService = this;
+        router.events
+            .pipe(
+                filter(x => x instanceof NavigationEnd),
+                map(() => activatedRoute.snapshot),
+                map(route => {
+                    while (route && !route.paramMap?.has('conferenceId')) {
+                        route = route?.firstChild;
+                    }
+
+                    return route?.paramMap;
+                })
+            )
+            .subscribe(paramMap => {
+                this.onRouteParamsChanged(paramMap);
+            });
+    }
+
+    private _currentConference: ConferenceResponse;
+    get currentConference(): ConferenceResponse {
+        return this._currentConference;
+    }
+
+    private currentConferenceSubject: ReplaySubject<ConferenceResponse> = new ReplaySubject<ConferenceResponse>();
```

Comment:
Should we not rely on implicit type here, I mean we don't really have to type cast a property if we are initializing it? 
So this would be the same? 
`private currentConferenceSubject = new ReplaySubject<ConferenceResponse>();`

Reply:
Yeah I will change this

Reply:
Any reason this is a ReplaySubject and not a BehaviorSubject?